Tella/Scenes/Settings/ViewModel/GDriveAuthViewModel.swift

```diff
@@ -12,21 +12,23 @@ import Combine
 class GDriveAuthViewModel: ObservableObject {
     private let gDriveRepository: GDriveRepositoryProtocol
     private var cancellables = Set<AnyCancellable>()
-
-    init(repository: GDriveRepositoryProtocol = GDriveRepository.shared) {
+    @Published var signInState: ViewModelState<String?> = .loaded(nil)
+    init(repository: GDriveRepositoryProtocol) {
```

Comment:
If we are not using the String in `ViewModelState` we can replace it with a boolean to indicate if the user is signed in or not

Reply:
I get your point but we are using the String to display the Toast of the error:

```
 catch let error {
                DispatchQueue.main.async {
                    self.signInState = .error(error.localizedDescription)
                }
            }
  ```
  
  I could create a custom type just to handle the signInState or to handle cases where the loaded state could be nil but I remember some month ago we talked about using just one generic type for managing states

Reply:
e.g for this i could create an enum SignInState:

```
enum SignInState: Equatable {
    case init
    case loading
    case loaded
    case failed(String)
}
```

for initialize:

```
@Published var signInState: SignInState = .notStarted
```

I think this could be a clearer approach but the drawback would be creating a new type only for this